rotkehlchen/globaldb/handler.py

```diff
@@ -1011,6 +1011,123 @@ def add_historical_prices(entries: List['HistoricalPrice']) -> None:
+            log.error(
+                f'One of the given historical price entries caused a DB error. {str(e)}. ',
+            )
+        connection.commit()
+
+    @staticmethod
+    def get_manual_prices(asset: 'Asset') -> List[Dict[str, str]]:
+        """Returns prices added to the database by the user for an asset"""
+        connection = GlobalDBHandler()._conn
+        cursor = connection.cursor()
+        querystr = (
+            'SELECT from_asset, to_asset, source_type, timestamp, price FROM price_history '
+            'WHERE from_asset=? AND source_type="A" ORDER BY timestamp'
+        )
+        query = cursor.execute(querystr, (asset.identifier,))
+        result = query.fetchall()
+        return [
+            {
+                'from_asset': entry[0],
+                'to_asset': entry[1],
+                'timestamp': entry[3],
+                'price': entry[4],
+            }
+            for entry in result
+        ]
+
+    def edit_manual_price(self, entry: HistoricalPrice) -> bool:
+        """Returns prices added to the database by the user for an asset"""
+        connection = GlobalDBHandler()._conn
+        cursor = connection.cursor()
+        # check that the entry we want to edit exitst
+        querystr = (
+            'SELECT COUNT(*) FROM price_history '
+            'WHERE from_asset=? AND to_asset=? AND timestamp=? AND source_type="A"'
+        )
+        params = (
+            entry.from_asset.identifier,
+            entry.to_asset.identifier,
+            entry.timestamp,
+        )
+        query = cursor.execute(querystr, params)
+        if query.fetchone()[0] == 0:
+            self.add_single_historical_price(entry)
+            return True
+
+        querystr = (
+            'UPDATE price_history SET price=? WHERE from_asset=? AND to_asset=? '
+            'AND timestamp=? AND source_type="A"'
+        )
+        entry_serialized = entry.serialize_for_db()
+        params_update = (
+            entry_serialized[4],
+            entry_serialized[0],
+            entry_serialized[1],
+            entry_serialized[3],
+        )
+        try:
+            cursor.execute(querystr, params_update)
+        except sqlite3.IntegrityError as e:
+            connection.rollback()
+            log.error(
+                f'Failed to edit manual historical prices from {entry.from_asset} '
+                f'to {entry.to_asset} at timestamp: {str(entry.timestamp)} due to {str(e)}',
+            )
+            return False
+        return True
+
+    @staticmethod
+    def delete_manual_price(
+            from_asset: 'Asset',
+            to_asset: 'Asset',
+            timestamp: Timestamp,
+    ) -> None:
+        connection = GlobalDBHandler()._conn
+        cursor = connection.cursor()
+        querystr = (
+            'DELETE FROM price_history WHERE from_asset=? AND to_asset=? '
+            'AND timestamp=? AND source_type="A"'
```

Comment:
make source type also a ?, and use the enum's serialize function.

Reply:
Here I don't have a `HistoricalPrice` instance to use the serialize. I'm missing fields here

Reply:
Please still do what I asked. You are hard coding a serialization value. I am asking you to add a `?` and then use the `HistoricalPrice.MANUAL.serialize_for_db()` to derive the `A`